### PR TITLE
Add better configuration methods to the builders

### PR DESCRIPTION
**Pull request by MinnDevelopment**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [x] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

This allows disabling flags without needing to specify an entire config.
This is very useful with the new factory methods.

~~We should also consider adding `enableIntents` and `disableIntents` for the same reason. I've seen users do `GatewayIntent.getIntents(DEFAULT & ~intents)` which would be a lot cleaner using `disableIntents(intents)`.~~ I decided to include this in the PR.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/api/JDABuilder.java [hunk 1]
```diff
@@ -477,14 +515,69 @@ public JDABuilder setRelativeRateLimit(boolean enable)
...
+     *
+     * @return The JDABuilder instance. Useful for chaining.
+     *
+     * @see    #disableCache(Collection)
+     */
+    @Nonnull
+    public JDABuilder enableCache(@Nonnull Collection<CacheFlag> flags)
+    {
+        Checks.noneNull(flags, "CacheFlags");
+        cacheFlags.removeAll(flags);
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/JDABuilder.java]:
```suggestion
        cacheFlags.addAll(flags);
```

File: src/main/java/net/dv8tion/jda/api/JDABuilder.java [hunk 2]
```diff
@@ -477,14 +515,69 @@ public JDABuilder setRelativeRateLimit(boolean enable)
...
+     *
+     * @throws IllegalArgumentException
+     *         If provided with null
+     *
+     * @return The JDABuilder instance. Useful for chaining.
+     *
+     * @see    #disableCache(CacheFlag, CacheFlag...)
+     */
+    @Nonnull
+    public JDABuilder enableCache(@Nonnull CacheFlag flag, @Nonnull CacheFlag... flags)
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/JDABuilder.java]: Would it be worth it to point this at the `Collections` variant method?

File: src/main/java/net/dv8tion/jda/api/sharding/DefaultShardManagerBuilder.java [hunk 3]
```diff
@@ -452,14 +490,69 @@ public DefaultShardManagerBuilder setRelativeRateLimit(boolean enable)
...
+     *
+     * @return The DefaultShardManagerBuilder instance. Useful for chaining.
+     *
+     * @see    #disableCache(Collection)
+     */
+    @Nonnull
+    public DefaultShardManagerBuilder enableCache(@Nonnull Collection<CacheFlag> flags)
+    {
+        Checks.noneNull(flags, "CacheFlags");
+        cacheFlags.removeAll(flags);
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/sharding/DefaultShardManagerBuilder.java]:
```suggestion
        cacheFlags.addAll(flags);
```

File: src/main/java/net/dv8tion/jda/api/JDABuilder.java (hunk 2 above)
Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/api/JDABuilder.java]: You mean an `@see`?

**Pull request merged**